### ISSUE TITLE
Gamma Correction applied twice with NVENC

### ISSUE THREAD

**Issue by Okabintaro** [Meumeu/WiVRn#74]
I've been experimenting extensively with WiVRn with nvenc (dev branch) today and managed to achieve quite a nice image quality. However, I've noticed an issue with the color contrast: light grays appear as white, and dark grays are nearly black. This suggests that Gamma Correction might be applied twice somewhere in the processing chain. I intend to investigate the nvenc code to address this, but I wanted to document this issue here as well.

**Comment by xytovl** [comment]
Have you confirmed that it is related to nvenc? If you configure encoder to x264, or even split between the 2, is there an image difference?

The ffmpeg based decoder (for Linux client) is known to do have incorrect gamma.

**Comment by Okabintaro** [comment]
Ok, I tested it against the x264 encoder and it seems that it is indeed related to nvenc.
I made some screenshots where I compared [this test image from wikipedia](https://upload.wikimedia.org/wikipedia/commons/e/e5/Gamma_correction_test_picture.png)

This is with **x264**: 

![test_image_x264](https://github.com/Meumeu/WiVRn/assets/103938900/7353577d-06ff-4a6f-bdf0-0073a986a627)

And here the same with **nvenc**:

![test_image_nvenc](https://github.com/Meumeu/WiVRn/assets/103938900/ba12e108-fcd1-444c-bd3b-6840cdde9a7e)

You can see the grays are washed out there more.
Here the config used for reference:

```json
{
  "scale": [
    0.7,
    0.7
  ],
  "bitrate": 50000000,
  "encoders": [
    {
      "encoder": "nvenc",
      "codec": "h264"
    }
  ],
  "tcp_only": true
}
```

**Comment by xytovl** [comment]
According to discussions on discord, the error is in full range vs limited range encoding.
Branch https://github.com/Meumeu/WiVRn/tree/fix/nvenc-color-scale contains a fix and should be validate before I merge it into dev.

**Comment by Okabintaro** [comment]
TGhe colors are fine in that branch, but it still crashes with sliced encoding on high scales, see #70

**Comment by xytovl** [comment]
Fixed on dev by 0d62a24f2593a6036c93e6d20d7e11f86117b066
